src/components/app/list-new/GenericAppList.tsx

```diff
@@ -101,184 +135,111 @@ const GenericAppList = ({
                     }
                 })
-
-                const recievedExternalFluxApps = externalAppData?.result?.fluxApplication || []
-                setAppsList((currAppList) => [...currAppList, ...recievedExternalFluxApps])
-            } catch (err) {
-                showError(err)
-                setErrorResponseCode(err.code)
-            } finally {
+                const receivedExternalFluxApps = externalAppData?.result?.fluxApplication || []
+                setAppsList((currAppList) => [...currAppList, ...receivedExternalFluxApps])
                 setDataStateType(AppListViewType.LIST)
+            } catch (err) {
+                setDataStateType(AppListViewType.ERROR)
             }
         }
-        _sseConnection.onerror = (err) => {
-            closeSseConnection(_sseConnection)
+        sseConnection.onerror = () => {
+            closeSseConnection(sseConnection)
         }
-        setSseConnection(_sseConnection)
+        setSseConnection(sseConnection)
     }
 
     // component load
     useEffect(() => {
         init()
-    }, [])
-
-    // filtering/sorting has been applied
-    useEffect(() => {
-        if (dataStateType === AppListViewType.LIST) {
-            if (clusterIdsCsv === _getClusterIdsFromRequestUrl()) {
-                handleFilteration()
-            } else {
-                init()
-            }
+        return () => {
+            setShowPulsatingDot(false)
         }
-    }, [payloadParsedFromUrl, dataStateType])
-
-    // when external app data comes
-    useEffect(() => {
-        if (dataStateType == AppListViewType.LIST) {
-            handleFilteration()
-        }
-    }, [appsList])
+    }, [])
 
     const handleArgoAppListing = () => {
         setDataStateType(AppListViewType.LOADING)
         getArgoInstalledExternalApps(clusterIdsCsv)
             .then((appsListResponse) => {
                 setAppsList(appsListResponse.result)
+                setDataStateType(AppListViewType.LIST)
             })
             .catch((errors: ServerErrors) => {
-                showError(errors)
                 setErrorResponseCode(errors.code)
-            })
-            .finally(() => {
-                setDataStateType(AppListViewType.LIST)
+                setDataStateType(AppListViewType.ERROR)
             })
     }
 
     const handleFluxAppListing = () => {
+        setAppsList([])
         setDataStateType(AppListViewType.LOADING)
         getExternalInstalledFluxApps(clusterIdsCsv)
     }
 
     useEffect(() => {
-        // fetch external apps list
-        if (isArgoCDAppList) {
-            handleArgoAppListing()
+        if (!clusterIdsCsv) {
+            setAppsList([])
+            setShowPulsatingDot(true)
             return
         }
-        if (isFluxCDAppList && clusterIdsCsv) {
-            handleFluxAppListing()
-        }
+        setShowPulsatingDot(false)
+        isArgoCDAppList ? handleArgoAppListing() : handleFluxAppListing()
```

Comment:
We should save the method in const variable and call that separately, this will fail in eslint check